### ISSUE TITLE
Reduce mentor timeout from 1 week to 3 days

### ISSUE THREAD

**Issue by Cohen-Carlisle** [exercism/exercism#4674]
I think a week is a very, very long time to have to wait for an unresponsive mentor. 3 days surely should be enough for a mentor that started a solution to get back to it?

**Comment by bitfield** [comment]
From the student's point of view, I definitely see that they don't want to wait a week _plus_ the time for another mentor to pick it up.

But from the mentor's point of view, I can also see where some mentors might only get a few hours a week for Exercism mentoring, and they would want to batch it. Like, spend three hours on a Sunday afternoon bombing through all the solutions in their queue. I'm conscious that we already ask a lot of mentors, and making mentoring a more demanding and time-critical process doesn't seem like a move in the right direction. For example, I'd love to work on open source projects (like Exercism) every day, but my schedule doesn't allow for it. Instead, I devote one afternoon a week to open source work.

Bouncing a solution from one mentor to another is also something I'd like to avoid wherever possible. It can be quite confusing for a student to have one mentor asking them to do one thing, and then all of a sudden there's a bunch of comments about something different from someone else.

Ideally, I don't really want any student to be waiting more than 24 hours to get feedback. If you're doing Exercism, you probably want to do it every day, and if you can't, it's hard to build that habit. So in that sense, three days is about three times too long. But some mentors may think it's too short!

It might be worth thinking about other ways to help manage the student experience, too. For example, we might add a 'Request a new mentor' button, so that if the student is in a big hurry and they feel like the mentor's taking too long, they can put the solution back in the queue themselves. We could show the student the average waiting time for solutions in this track, or for this mentor. I've heard several good ideas on this topic, many of which are less of a blunt instrument than kicking the mentor off after a fixed time.

**Comment by tehsphinx** [comment]
@bitfield basically nailed it.

I agree: A week can be very long for a student. But at the same time very short for a mentor. I think it would add more pressure on a mentor which we shouldn't do.

**Comment by NobbZ** [comment]
As a student I have already to wait for a very long time to get the first review. If I then have to wait a week + another very long time to get the second opinion, this could sum up to months very quickly, especially in the tracks that already are at their limits, eg: JS, Rust, python…

As a mentor, I usually mentor one or two times a week in batches, usually on mondays, sometimes thursdays, at least if I have some time-off in the office. I've stopped mentoring at home. When a student replies though, I usually get an email and will act on the next **office day** as early as possible. So if I get notified on a friday, I wont get back to it before monday, thats exactly 3 days. In my experience, even when I bulk mentor 10+ solutions, the students reactions will spread over the next 5 days from alone. And replying those takes much less time than the initial review.

So from my point of view 3 days is the bare minimum, while I suggest as well to not keep the timeout at a whole week.

**Comment by yawpitch** [comment]
I'm just going to chime in and say that a week is indeed a long time to wait for an _unresponsive_ mentor ... but that a week is just about the right amount of time to wait before deciding someone _is_ unresponsive.

I'm not even working right now and I don't have sufficient time to work on this every day; I've got to batch them up and might only get 1-2 chances a week to dedicate to mentoring. I can only imagine that it's a much greater problem for someone with a full-time job as well.

For me 3 days would be at least 2 days too few to be a reasonable minimum. Any shorter than 5 days and you'll definitely end up losing me.

**Comment by alanmynah** [comment]
Feels like the core issue we're trying to address is students being blocked by feedbacks. Please correct me if I'm wrong. 

Is there a technical possibility to reduce the timeout for some exercises and not for others? 
From mentors point of view batching is important and most comfortable to do once a week. 
From students point of view - the first, say 5-10 exercises, are critical, because that's when they run out of exercises to do on any given track, as they get blocked by reviews. Afterwards, there are enough side exercises unlocked to keep oneself busy.

**Comment by iHiD** [comment]
> From students point of view - the first, say 5-10 exercises, are critical, because that's when they run out of exercises to do on any given track, as they get blocked by reviews. Afterwards, there are enough side exercises unlocked to keep oneself busy.

For tracks where this is true, the tracks need improving. Each core exercise should unlock 5-10 side exercises so the learner shouldn't run out.

I'm not against the idea of variable timeouts though.

**Comment by iHiD** [comment]
> For example, we might add a 'Request a new mentor' button, so that if the student is in a big hurry and they feel like the mentor's taking too long, they can put the solution back in the queue themselves. We could show the student the average waiting time for solutions in this track, or for this mentor. I've heard several good ideas on this topic, many of which are less of a blunt instrument than kicking the mentor off after a fixed time.

I agree with this that there are less "blunt" changes we can make in the long-run to address this problem.

So, there's currently a prompt at 5 days and a timeout at 7 days. There's seems to be a **loose** consensus that changing that to prompt at 3 days and timeout at 5 days would be generally supported. Is that a fair reading of this thread so far?

**Comment by Smarticles101** [comment]
How bout we just delete the website altogether

**Comment by Smarticles101** [comment]
What if it was up to mentors to decide. They're solutions time out at x amount of days, 1 week being the most? Students could be able see a mentors timeout time.

**Comment by son1112** [comment]
Throwing my 2 cents in without yet reading everyone's comments yet, so I apologize if there is any repetition in this. 

I just recently was approved to spend 10% of my work at CardFlight contributing to Exercism, as a mentor and to the CLI. The scheduled time is every other Friday. I love Exercism to bits and will 
be, more often than not, contributing and participating in my free time/breaks during regular work hours as well. However, many weeks at work are fairly busy, so some of those my time only be one day every two weeks.

**Comment by son1112** [comment]
@bitfield I like the 'Request a new mentor' button idea

**Comment by yawpitch** [comment]
> So, there's currently a prompt at 5 days and a timeout at 7 days. There's seems to be a **loose** consensus that changing that to prompt at 3 days and timeout at 5 days would be generally supported. Is that a fair reading of this thread so far?

I'd be much more inclined to move the prompt to 3 and the timeout to 6.

**Comment by tehsphinx** [comment]
What is meant with "prompt"? If that is an email: I have switched emails off, so there will be no prompt!

Also I like the idea more of adding a `Request a new mentor` button than narrowing the window for mentors.

To add to that idea: Don't show the button before 3 or 4 days have passed. Then show the button with a note saying that the mentor will be auto-removed after x more days. (x being 7 - passed days).

**Comment by mstange22** [comment]
Hi all - not sure if this is the right place to post, but I thought I would try here before opening a new issue.  I think there might be a problem with one of my submissions that has received initial mentor feedback.  It has been 9 days since initial mentor response and resubmittal and there was a curious circumstance immediately after I submitted my revision.

The exercise is C | Phone Number: https://exercism.io/my/solutions/a1c9c98c43cd4c65bf507056b2a4da77

After initial mentor comments, I made the requested changes and resubmitted within 5-6 hours.  A few hours later I received an email notifying me that I still have a pending solution and should address the mentor feedback (that I already addressed).  This is the first time I have received any email like this (maybe because I'm usually relatively quick at getting my revisions submitted?).

Anyway, it's been almost 10 days now and still no second response on my revision.  I'm wondering if there was some sort of issue since I received that odd email and my submission was not actually received?

**Comment by ErikSchierboom** [comment]
Hello 👋

With the launch of Exercism v3, we are **closing all issues in this repository** to help give us a clean slate to detect new problems. If this issue is still relevant to Exercism v3 (e.g. it's a feature that we haven't implemented in v3, or a bug that still exists), please reopen it and we will review it and post an update on it as soon as we get chance.

Thanks for helping make Exercism better, and we hope you enjoy v3 🙂
